streams/src/main/java/org/apache/kafka/streams/query/QueryResult.java

```diff
@@ -197,6 +197,18 @@ public R getResult() {
         return result;
     }
 
+    @SuppressWarnings("unchecked")
+    public <V> QueryResult<V> swapResult(final V value) {
```

Comment:
We don't really need this. You can just do `QueryResult.forResult` in the `MeteredKeyValue` store for example to get the typed result

Reply:
The purpose of this method is to allow `MeteredKeyValue` store to deserialize the result without wiping out the execution info or position that it got back from the bytes store. I missed that while reviewing your PR, so I went ahead and added a fix for it to this one.